src/Helpers.js

```diff
@@ -321,16 +323,21 @@ export let readCsv = (pathname, delimiter = ',') => {
   // Read the CSV file at the pathname and parse it with the given delimiter.
   // Return an array of rows that are each an array of columns.
   // If there is a header row, it is returned as the first row.
-  let result = null
-  const CSV_PARSE_OPTS = {
-    delimiter,
-    columns: false,
-    trim: true,
-    skip_empty_lines: true,
-    skip_lines_with_empty_values: true
+  // Cache parsed files to support multiple reads from different equations.
+  let csv = csvData.get(pathname)
+  if (csv == null) {
+    const CSV_PARSE_OPTS = {
+      delimiter,
+      columns: false,
+      trim: true,
+      skip_empty_lines: true,
+      skip_lines_with_empty_values: true
+    }
+    let data = B.read(pathname)
+    csv = parseCsv(data, CSV_PARSE_OPTS)
+    csvData.set(pathname, csv)
```

Comment:
I wonder if we need to worry about caching many large files in memory being an issue for models that read many large CSV files (but that only make use of parts of them).  But I suspect the answer is "probably not", and if we do ever encounter such a beast, we can worry about it at that time.